### ISSUE TITLE
Custom parsii functions

### ISSUE THREAD

**Issue by dfsek** [PolyhedralDev/Terra#69]
Add a way to register custom Parsii functions in the pack manifest, probably under a `functions` key analogous to variables.

Functions would have a customizable number of arguments, represented as alphabetic variables within a parsii equation, e.g. 3 arguments = `a` `b` and `c` as vars within the equation.

Mockup:
```yml
functions:
  fourMax:
    arguments: 4
    function: "max(max(max(a, b), c), d)"

```

These would be parsed per-biome, and registered separately to each biome's Expression instances, to allow them to
* Use overridden variables in biomes
* Be overridden in biomes.

**Comment by dfsek** [comment]
Maybe also optionally add a way to give argument variables custom names, e.g.
```yml
functions:
  fourMax:
    arguments: 4
    variable-names:
      - w
      - x
      - y
      - z
    function: "max(max(max(w, x), y), z)"
```

Any indices not present in the names list would just be assigned alphabetic names.

**Comment by dfsek** [comment]
Implemented in 1863ba29fc68ff31e377bc92e72123f2d65bbc31
